src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,290 @@
 import { IQuerySuggestSelection } from '../../events/OmniboxEvents';
+import { IQueryResults } from '../../rest/QueryResults';
+import 'styling/_QuerySuggestPreview';
+import { ResultListRenderer } from '../ResultList/ResultListRenderer';
+import { IInitializationParameters, IInitResult } from '../Base/Initialization';
+import { IResultListOptions } from '../ResultList/ResultListOptions';
+import { Template } from '../Templates/Template';
+import { TemplateComponentOptions } from '../Base/TemplateComponentOptions';
+import { IQueryResult } from '../../rest/QueryResult';
+import { pluck, sortBy, map } from 'underscore';
+import { ResultListTableRenderer } from '../ResultList/ResultListTableRenderer';
 
-export interface IQuerySuggestPreview extends IResultListOptions {
+export interface IQuerySuggestPreview {
   numberOfPreviewResults?: number;
-  width?: string;
+  previewWidth?: number;
+  suggestionWidth?: string;
+  resultTemplate?: Template;
+  headerText?: string;
+  hoverTime?: number;
 }
 
-export class QuerySuggestPreview extends ResultList implements IComponentBindings {
+/**
+ * This component need [`querySuggest`]{@link Omnibox.options.enableQuerySuggestAddon} to work.
+ *
+ *  It allows you to preview the top recommendation when hovering on a Query suggest
+ *
+ *  A [Result Templates](https://developers.coveo.com/x/aIGfAQ) is required for this component to work.
+ */
+export class QuerySuggestPreview extends Component implements IComponentBindings {
   static ID = 'QuerySuggestPreview';
 
-  static elementsToIgnore = [];
-
   static doExport = () => {
     exportGlobally({
       QuerySuggestPreview: QuerySuggestPreview
     });
   };
 
+  /**
+   * The options for the component
+   * @componentOptions
+   */
   static options: IQuerySuggestPreview = {
-    numberOfPreviewResults: ComponentOptions.buildNumberOption({ defaultValue: 0, min: 0 })
+    resultTemplate: TemplateComponentOptions.buildTemplateOption(),
+    /**
+     * The maximum number of query results to render in the preview.
+     *
+     * **Minimum value:** `1`
+     * **Maximum value:** `6`
+     * **Default value:** `3`
+     */
+    numberOfPreviewResults: ComponentOptions.buildNumberOption({
+      defaultValue: 3,
+      min: 1,
+      max: 6
+    }),
+    /**
+     *  Width in `pixels` that the preview container will take
+     *
+     * The default behavior is to take 100% of the remaining place shared by the suggestion
+     */
+    previewWidth: ComponentOptions.buildNumberOption(),
+    /**
+     *  Width in `pixels` or `pourcentages` that the suggestion container will take
+     *
+     * **Default value:** `33%`
+     */
+    suggestionWidth: ComponentOptions.buildStringOption({ defaultValue: '33%' }),
+    /**
+     *  The text displayed at the top of the preview.
+     *
+     *  After this text, the suggestion will be surrounded in quotes
+     *
+     * **Default value:** `Product recommandation for`
```

Comment:
It may not be a `product` recommendation. The item may be an email, a KB article, etc. So I'd make the default value something like:
`Query result items for`